pymove/tests/core/test_data_frame.py

```diff
@@ -0,0 +1,1102 @@
+
+
+def test_join():
+    move_df = _default_move_df()
+    other = DataFrame({'id': [1, 1, 2, 2], 'key': ['a', 'b', 'c', 'd']})
+
+    result = move_df.join(other, on='id', lsuffix='_l', rsuffix='_r')
+    expected = DataFrame(data=[
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1, 1, 'b'],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1, 1, 'b'],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, 2, 'c'],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, 2, 'c'],
+    ], columns=['lat', 'lon', 'datetime', 'id_l', 'id_r', 'key'], index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(result, expected)
+
+
+def test_merge():
+    move_df = _default_move_df()
+    other = DataFrame({'key': [1], 'value': ['a']})
+
+    result = move_df.merge(other, left_on='id', right_on='key')
+    expected = DataFrame(data=[
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1, 1, 'a'],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1, 1, 'a'],
+    ], columns=['lat', 'lon', 'datetime', 'id', 'key', 'value'], index=[0, 1]
+    )
+    assert_frame_equal(result, expected)
+
+
+def test_nunique():
+    move_df = _default_move_df()
+    assert_array_equal(move_df.nunique(), [3, 3, 3, 2])
+
+
+def test_write_file(tmpdir):
+    d = tmpdir.mkdir('prepossessing')
+
+    file_write_default = d.join('test_write_default.csv')
+    filename_write_default = os.path.join(
+        file_write_default.dirname, file_write_default.basename
+    )
+    move_df = _default_move_df()
+    move_df.write_file(filename_write_default)
+    move_df_test = read_csv(filename_write_default)
+    assert_frame_equal(move_df, move_df_test)
+
+    file_write_semi = d.join('test_write_semi.csv')
+    filename_write_semi = os.path.join(
+        file_write_semi.dirname, file_write_semi.basename
+    )
+    move_df = _default_move_df()
+    move_df.write_file(filename_write_semi, separator=';')
+    move_df_test = read_csv(filename_write_semi, sep=';')
+    assert_frame_equal(move_df, move_df_test)
+
+
+def test_to_csv(tmpdir):
+    d = tmpdir.mkdir('prepossessing')
+
+    file_csv = d.join('test_csv.csv')
+    filename_csv = os.path.join(
+        file_csv.dirname, file_csv.basename
+    )
+    move_df = _default_move_df()
+    move_df.to_csv(filename_csv, index=False)
+    move_df_test = read_csv(filename_csv)
+    assert_frame_equal(move_df, move_df_test)
+
+
+def test_convert_to():
+    move_df = _default_move_df()
+
+    assert move_df._type == TYPE_PANDAS
+    assert isinstance(move_df, PandasMoveDataFrame)
+    assert isinstance(move_df._data, DataFrame)
+
+    move_df_dask = move_df.convert_to('dask')
+    assert move_df_dask._type == TYPE_DASK
+    assert isinstance(move_df_dask, DaskMoveDataFrame)
+    assert isinstance(move_df_dask, DaskDataFrame)
+
+
+def test_get_type():
+    move_df = _default_move_df()
+
+    assert move_df.get_type() == TYPE_PANDAS
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574968)